For the statistical calculation of the t-test(a significant difference in the categories) just for practice, can maybe use both z and t test (should be able to derive the same ans)

In [2]:
import numpy as np
import pandas as pd
from statsmodels.stats import weightstats 
import scipy.stats as stats

In [3]:
transac_data = pd.read_csv('Dummy Data (3 months).csv')

In [4]:
transac_data.head(10)

Date Description of Purchase Category of Purchase  Price 
0   1/1/22           Grocery Store            Groceries   45.67
1   2/1/22             Gas Station       Transportation   35.12
2   3/1/22             Coffee Shop           Dining Out    8.99
3   4/1/22           Movie Theater        Entertainment   22.50
4   5/1/22              Amazon.com      Online Shopping   67.89
5   6/1/22              Hair Salon        Personal Care   55.00
6   7/1/22          Gym Membership       Health/Fitness   80.00
7   8/1/22          Clothing Store             Clothing  112.45
8   9/1/22           Grocery Store            Groceries   23.12
9  10/1/22             Gas Station       Transportation   40.23

In [24]:
#getting rid of whitespace in the headers
transac_data.columns = transac_data.columns.str.strip()

In [25]:
#changing the data types
#1: date to dmy
transac_data['Date'] = pd.to_datetime(transac_data['Date'], format = '%d/%m/%y')

In [26]:
#2: description of purchase and category of purchase as string
transac_data[['Description of Purchase', 'Category of Purchase']] = transac_data[['Description of Purchase', 'Category of Purchase']].astype('string')

**Analysis 1**: Any significant difference between the amount spent within the different categories

Categories: Groceries, Transportation, Dining Out, Entertainment, Online Shopping, Personal Care 

In [6]:
#filtered item for groceries
groceries = transac_data[transac_data['Category of Purchase']== 'Groceries']['Price']
groceries

0     45.67
8     23.12
16    41.23
24    27.23
30    42.89
38    23.56
46    37.23
54    29.23
58    35.67
66    28.34
74    31.23
82    26.23
Name: Price, dtype: float64

In [7]:
#filtered item for Transportation
transportation = transac_data[transac_data['Category of Purchase']== 'Transportation']['Price']
transportation

1     35.12
9     40.23
17    37.12
25    38.23
31    30.12
39    40.23
47    35.12
55    38.23
59    25.42
67    36.12
75    29.45
83    39.56
Name: Price, dtype: float64

In [8]:
#filtered item for Dining Out
dining_out = transac_data[transac_data['Category of Purchase']== 'Dining Out']['Price']
dining_out

2      8.99
10    12.99
18     9.99
26    10.99
32     8.50
40    12.99
48     9.99
56    10.99
60     7.99
68    10.99
76     8.50
84    12.99
Name: Price, dtype: float64

In [9]:
#filtered item for Entertainment
entertainment = transac_data[transac_data['Category of Purchase']== 'Entertainment']['Price']
entertainment

3     22.50
11    28.75
19    24.50
27    26.75
33    25.00
41    28.50
49    22.75
57    26.75
61    20.50
69    23.75
77    27.50
85    21.75
Name: Price, dtype: float64

In [10]:
#filtered item for Online Shopping
online_shop = transac_data[transac_data['Category of Purchase'] == 'Entertainment']['Price']
online_shop

3     22.50
11    28.75
19    24.50
27    26.75
33    25.00
41    28.50
49    22.75
57    26.75
61    20.50
69    23.75
77    27.50
85    21.75
Name: Price, dtype: float64

In [11]:
#filtered item for Personal Care
personal_care = transac_data[transac_data['Category of Purchase'] == 'Personal Care']['Price']
personal_care

5     55.0
13    67.0
21    62.0
29    71.0
35    65.0
43    72.0
51    60.0
63    62.0
71    75.0
79    68.0
87    80.0
Name: Price, dtype: float64

----------------------------------------------------------------------------------------------------------------------

In [12]:
#conducting t-test for the different categories
#groceries and transportation
stats.ttest_ind(groceries, transportation, equal_var=True)

Ttest_indResult(statistic=-1.062842990461972, pvalue=0.2993811934896691)

p-value = 0.299 > 0.05. We do not reject the null hypothesis. There is no significant difference in the spendings for groceries and transportation.

In [14]:
#groceries and dining out
stats.ttest_ind(groceries, dining_out, equal_var = True)

Ttest_indResult(statistic=9.679605925063399, pvalue=2.175878554643374e-09)

p-value = 2.175878554643374e-09 < 0.05. We reject null hypothesis. There is a significant difference in the spendings for groceries and dining out.

In [23]:
#groceries and entertainment
stats.ttest_ind(groceries, entertainment, equal_var = True)

Ttest_indResult(statistic=3.266894307563918, pvalue=0.0035282759831674008)

p-value = 0.0035283 < 0.05. We reject the null hypothesis. There is significant difference in the spendings for groceries and entertainment.

In [25]:
#groceries and online shopping
stats.ttest_ind(groceries, online_shop, equal_var = True)

Ttest_indResult(statistic=3.266894307563918, pvalue=0.0035282759831674008)

p-value = 0.0035283 < 0.05. We reject the null hypothesis. There is significant difference in the spendings for groceries and entertainment.

In [26]:
#groceries and personal care
stats.ttest_ind(groceries, personal_care, equal_var = True)

Ttest_indResult(statistic=-10.99910646742078, pvalue=3.5653466492395016e-10)

p-value = 3.5653466492395016e-10 < 0.05. We reject the null hypothesis. There is a significant difference in the spendings for groceries and personal care. 

In [28]:
#transportation and dining out
stats.ttest_ind(transportation, dining_out, equal_var = True)

Ttest_indResult(statistic=17.02244760743162, pvalue=3.7574222893432807e-14)

p-value = 3.7574222893432807e-14. We reject the null hypothesis. There is a significant difference in the spendings for transportation and dining out.

In [30]:
#transportation and entertainment
stats.ttest_ind(transportation, entertainment, equal_var = True)

Ttest_indResult(statistic=6.647865962628232, pvalue=1.1029140191760613e-06)

p-value = 1.1029140191760613e-06 < 0.05. We reject the null hypothesis. There is a significant difference in the spendings for transportation and entertainment. 

In [31]:
#transportation and online shopping
stats.ttest_ind(transportation, online_shop, equal_var = True)

Ttest_indResult(statistic=6.647865962628232, pvalue=1.1029140191760613e-06)

p-value = 1.1029140191760613e-06 < 0.05. We reject the null hypothesis. There is a significant difference in the spendings for transportation and online shopping.

In [32]:
#transportation and personal care
stats.ttest_ind(transportation, personal_care, equal_var = True)

Ttest_indResult(statistic=-12.509145549519868, pvalue=3.371108718508917e-11)

p-value = 3.371108718508917e-11 < 0.05. We reject the null hypothesis. There is significant difference in the spendings for transportation and online shopping.

In [33]:
#dining out and entertainment
stats.ttest_ind(dining_out, entertainment, equal_var = True)

Ttest_indResult(statistic=-15.194939077754787, pvalue=3.7872452230425636e-13)

p-value = 3.7872452230425636e-13 < 0.05. We reject the null hypothesis. There is significant difference in the spendings for dining out and entertainment.

In [34]:
#dining out and online shopping
stats.ttest_ind(dining_out, online_shop, equal_var = True)

Ttest_indResult(statistic=-15.194939077754787, pvalue=3.7872452230425636e-13)

p-value = 3.7872452230425636e-13 < 0.05. We reject the null hypothesis. There is significant difference in the spendings for dining out and online shopping.

In [35]:
#dining out and personal care
stats.ttest_ind(dining_out, personal_care, equal_var = True)

Ttest_indResult(statistic=-26.252136052320694, pvalue=1.5331153801767453e-17)

p-value = 1.5331153801767453e-17 < 0.05. We reject the null hypothesis. There is significant difference in the spendings for dining out and personal care.

In [36]:
#entertainment and online shopping
stats.ttest_ind(entertainment, online_shop, equal_var = True)

Ttest_indResult(statistic=0.0, pvalue=1.0)

????

In [38]:
#entertainment and personal care
stats.ttest_ind(entertainment, personal_care, equal_var = True)

Ttest_indResult(statistic=-18.790126657586825, pvalue=1.301376376101303e-14)

p-value = 1.301376376101303e-14 < 0.05. We reject the null hypothesis. There is significant difference in the spendings for entertainment and personal care.

In [39]:
#online shopping and personal care
stats.ttest_ind(online_shop, personal_care, equal_var = True)

Ttest_indResult(statistic=-18.790126657586825, pvalue=1.301376376101303e-14)

p-value = 1.301376376101303e-14 < 0.05. We reject the null hypothesis. There is significant difference in the spendings for entertainment and personal care.

**Analysis 2:** Calculating confidence interval (for each category of spending)

In [17]:
#groceries
mean = np.mean(groceries)
mean

32.63583333333333

In [18]:
sd = np.std(groceries)
sd

7.384778780633099

In [23]:
stats.t.interval(0.95, len(groceries) - 1, loc = mean, scale = sd / np.sqrt(len(groceries)))

(27.94376874849506, 37.3278979181716)

95% confident that the expenditures for groceries is (27.9438, 37.3279)

Groceries, Transportation, Dining Out, Entertainment, Online Shopping, Personal Care

In [21]:
#transportation
mean_t = np.mean(transportation)
mean_t

35.4125

In [22]:
sd_t = np.std(transportation)
sd_t

4.5322384554066275

In [27]:
stats.t.interval(0.95, len(transportation) - 1, loc = mean_t, scale = sd_t / np.sqrt(len(transportation)))

(32.53285306969176, 38.29214693030824)

95% confident the expenditures for groceries is (32.5329, 38.29215).

In [15]:
#Dining out
mean_dining = np.mean(dining_out)
mean_dining

10.491666666666665

In [16]:
sd_dining = np.std(dining_out)
sd_dining

1.7421291252053877

In [17]:
stats.t.interval(0.95, len(dining_out) - 1, loc = mean_dining, scale = sd_dining/np.sqrt(len(dining_out)))

(9.384770629099688, 11.598562704233643)

95% confident the expenditures for dining out is (34.3056, 36.5194).

In [32]:
#entertainment
mean_entertainment = np.mean(entertainment)
mean_entertainment

24.916666666666668

In [33]:
sd_entertainment = np.std(entertainment)
sd_entertainment

2.6226841636427025

In [34]:
stats.t.interval(0.95, len(entertainment) - 1, loc = mean_entertainment, scale = sd_entertainment/np.sqrt(len(entertainment)))

(23.250292649652565, 26.58304068368077)

95% confident that expenditures for entertainment is (23.25029, 26.58304).

In [35]:
#online shopping
mean_shop = np.mean(online_shop)
mean_shop

24.916666666666668

In [37]:
sd_shop = np.std(online_shop)
sd_shop

2.6226841636427025

In [38]:
stats.t.interval(0.95, len(online_shop) - 1, loc = mean_shop, scale = sd_shop/np.sqrt(len(online_shop)))

(23.250292649652565, 26.58304068368077)

95% confident that the expenditures for online shopping is (23.2503, 26.583).

In [39]:
#personal care
mean_care = np.mean(personal_care)
mean_care

67.0

In [40]:
sd_care = np.std(personal_care)
sd_care

6.888725967444594

In [43]:
stats.t.interval(0.95, len(personal_care) - 1, loc = mean_care, scale = sd_care/np.sqrt(len(personal_care)))

(62.37209092411135, 71.62790907588865)

95% confident that the expenditures for personal care is (62.3721, 71.62791).

**Analysis 3:** Any significant difference between the amount spent within the different months

In [37]:
transac_data.head(20)

Date Description of Purchase Category of Purchase   Price Month
0  2022-01-01           Grocery Store            Groceries   45.67   Jan
1  2022-01-02             Gas Station       Transportation   35.12   Jan
2  2022-01-03             Coffee Shop           Dining Out    8.99   Jan
3  2022-01-04           Movie Theater        Entertainment   22.50   Jan
4  2022-01-05              Amazon.com      Online Shopping   67.89   Jan
5  2022-01-06              Hair Salon        Personal Care   55.00   Jan
6  2022-01-07          Gym Membership       Health/Fitness   80.00   Jan
7  2022-01-08          Clothing Store             Clothing  112.45   Jan
8  2022-01-09           Grocery Store            Groceries   23.12   Jan
9  2022-01-10             Gas Station       Transportation   40.23   Jan
10 2022-01-11             Coffee Shop           Dining Out   12.99   Jan
11 2022-01-12           Movie Theater        Entertainment   28.75   Jan
12 2022-01-13              Amazon.com      Online Shopping   45.67   Jan
13 2022-01-14              Hair Salon        Personal Care   67.00   Jan
14 2022-01-15          Gym Membership       Health/Fitness   75.00   Jan
15 2022-01-16          Clothing Store             Clothing   89.99   Jan
16 2022-01-17           Grocery Store            Groceries   41.23   Jan
17 2022-01-18             Gas Station       Transportation   37.12   Jan
18 2022-01-19             Coffee Shop           Dining Out    9.99   Jan
19 2022-01-20           Movie Theater        Entertainment   24.50   Jan

In [33]:
#new 'month' column to group transactions in months
transac_data['Month'] = transac_data['Date'].dt.strftime('%b')
transac_data

Date Description of Purchase Category of Purchase  Price Month
0  2022-01-01           Grocery Store            Groceries  45.67   Jan
1  2022-01-02             Gas Station       Transportation  35.12   Jan
2  2022-01-03             Coffee Shop           Dining Out   8.99   Jan
3  2022-01-04           Movie Theater        Entertainment  22.50   Jan
4  2022-01-05              Amazon.com      Online Shopping  67.89   Jan
..        ...                     ...                  ...    ...   ...
83 2022-03-26             Gas Station       Transportation  39.56   Mar
84 2022-03-27             Coffee Shop           Dining Out  12.99   Mar
85 2022-03-28           Movie Theater        Entertainment  21.75   Mar
86 2022-03-29              Amazon.com      Online Shopping  36.99   Mar
87 2022-03-30              Hair Salon        Personal Care  80.00   Mar

[88 rows x 5 columns]

In [34]:
january = transac_data[transac_data['Month'] == 'Jan']['Price']
january

0      45.67
1      35.12
2       8.99
3      22.50
4      67.89
5      55.00
6      80.00
7     112.45
8      23.12
9      40.23
10     12.99
11     28.75
12     45.67
13     67.00
14     75.00
15     89.99
16     41.23
17     37.12
18      9.99
19     24.50
20     78.90
21     62.00
22     85.00
23    102.45
24     27.23
25     38.23
26     10.99
27     26.75
28     55.67
29     71.00
Name: Price, dtype: float64

In [39]:
february = transac_data[transac_data['Month'] == 'Feb']['Price']
february

30     42.89
31     30.12
32      8.50
33     25.00
34     68.99
35     65.00
36     90.00
37    110.45
38     23.56
39     40.23
40     12.99
41     28.50
42     46.67
43     72.00
44     80.00
45     89.99
46     37.23
47     35.12
48      9.99
49     22.75
50     49.99
51     60.00
52     85.00
53    102.45
54     29.23
55     38.23
56     10.99
57     26.75
Name: Price, dtype: float64

In [40]:
march = transac_data[transac_data['Month'] == 'Mar']['Price']
march

58     35.67
59     25.42
60      7.99
61     20.50
62     56.78
63     62.00
64     85.00
65     99.95
66     28.34
67     36.12
68     10.99
69     23.75
70     45.99
71     75.00
72     90.00
73    110.45
74     31.23
75     29.45
76      8.50
77     27.50
78     62.99
79     68.00
80     95.00
81    112.99
82     26.23
83     39.56
84     12.99
85     21.75
86     36.99
87     80.00
Name: Price, dtype: float64

In [43]:
#t-test for jan and feb
stats.ttest_ind(january, february, equal_var = True)

Ttest_indResult(statistic=0.1874935319157287, pvalue=0.8519513883887918)

p-value = 0.85195 > 0.05. We do not reject the null hypothesis. There is no significant difference between spendings in January and February.

In [44]:
#t test for jan and mar
stats.ttest_ind(january, march, equal_var = True)

Ttest_indResult(statistic=0.06100748277982143, pvalue=0.9515629896291878)

p-value = 0.95156 > 0.05. We reject the null hypothesis. There is no significant difference between spendings in January and March.

In [45]:
#t test for feb and mar
stats.ttest_ind(february, march, equal_var = True)

Ttest_indResult(statistic=-0.11790105995303116, pvalue=0.9065679300756864)

p-value = 0.90657 > 0.05. We reject the null hypothesis. There is no significant difference between spendings in February and March.